services/coverity/coverity-scan.service.js

```diff
@@ -1,63 +1,79 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
+const Joi = require('joi')
+const BaseJsonService = require('../base-json')
+
+const messageRegex = /passed|passed .* new defects|pending|failed/
+const schema = Joi.object({
+  message: Joi.string()
+    .regex(messageRegex)
+    .required(),
+}).required()
+
+module.exports = class CoverityScan extends BaseJsonService {
+  static render({ message }) {
+    let color
+    if (message === 'passed') {
+      color = 'brightgreen'
+      message = 'passing'
+    } else if (/^passed .* new defects$/.test(message)) {
+      color = 'yellow'
+    } else if (message === 'pending') {
+      color = 'orange'
+    } else {
+      color = 'red'
+    }
+
+    return {
+      message,
+      color,
+    }
+  }
 
-module.exports = class CoverityScan extends LegacyService {
   static get category() {
     return 'quality'
   }
 
+  static get defaultBadgeData() {
+    return {
+      label: 'coverity',
+    }
+  }
+
   static get route() {
     return {
       base: 'coverity/scan',
+      pattern: ':projectId',
     }
   }
 
   static get examples() {
     return [
       {
         title: 'Coverity Scan',
-        previewUrl: '3997',
+        pattern: ':projectId',
+        namedParams: {
+          projectId: '3997',
+        },
+        staticPreview: this.render({
+          message: 'passed',
+        }),
+        keywords: ['coverity', 'scan'],
```

Comment:
we don't need to specify either of these as keywords because they both appear in the title

Reply:
👍 I didn't realize that, good to know!